test/e2e/externally_provisioned_test.go

```diff
@@ -103,18 +104,211 @@ var _ = Describe("Create as externally provisioned, deprovision", Label("require
+			By("Creating a BMH without ExternallyProvisioned to allow inspection")
+			bmh := metal3api.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      specName,
+					Namespace: namespace.Name,
+				},
+				Spec: metal3api.BareMetalHostSpec{
+					Online: true,
+					BMC: metal3api.BMCDetails{
+						Address:                        bmc.Address,
+						CredentialsName:                secretName,
+						DisableCertificateVerification: bmc.DisableCertificateVerification,
+					},
+					BootMACAddress:        bmc.BootMacAddress,
+					ExternallyProvisioned: false, // Start with false to allow inspection
+				},
+			}
+			err := clusterProxy.GetClient().Create(ctx, &bmh)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the BMH to complete inspection and reach Available state")
+			WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+				Client: clusterProxy.GetClient(),
+				Bmh:    bmh,
+				State:  metal3api.StateAvailable,
+			}, e2eConfig.GetIntervals(specName, "wait-available")...)
+
+			By("Retrieving the BMH to verify inspection completed")
+			err = clusterProxy.GetClient().Get(ctx, types.NamespacedName{
+				Name:      bmh.Name,
+				Namespace: bmh.Namespace,
+			}, &bmh)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Verifying that inspection was performed")
+			Expect(bmh.Status.OperationHistory.Inspect.Start.IsZero()).To(BeFalse(), "Inspection should have been performed")
+			Expect(bmh.Status.HardwareDetails).NotTo(BeNil(), "Hardware details should be available after inspection")
+
+			By("Setting ExternallyProvisioned to true from Available state (allowed by webhook)")
+			patch := client.MergeFrom(bmh.DeepCopy())
+			bmh.Spec.ExternallyProvisioned = true
+			err = clusterProxy.GetClient().Patch(ctx, &bmh, patch)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the BMH to transition to ExternallyProvisioned state")
+			WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+				Client: clusterProxy.GetClient(),
+				Bmh:    bmh,
+				State:  metal3api.StateExternallyProvisioned,
+			}, e2eConfig.GetIntervals(specName, "wait-externally-provisioned")...)
+
+			By("Verifying that no BMO provisioning occurred after transition")
+			err = clusterProxy.GetClient().Get(ctx, types.NamespacedName{
+				Name:      bmh.Name,
+				Namespace: bmh.Namespace,
+			}, &bmh)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bmh.Status.OperationHistory.Provision.Start.IsZero()).To(BeTrue(), "BMO provisioning should not have occurred")
+
+			By("Cleaning up the BMH")
+			err = clusterProxy.GetClient().Delete(ctx, &bmh)
+			Expect(err).NotTo(HaveOccurred())
+
+			WaitForBmhDeleted(ctx, WaitForBmhDeletedInput{
+				Client:    clusterProxy.GetClient(),
+				BmhName:   bmh.Name,
+				Namespace: bmh.Namespace,
+			}, e2eConfig.GetIntervals(specName, "wait-bmh-deleted")...)
+		})
+
+		It("blocks enabling ExternallyProvisioned when not in Available state", func() {
```

Comment:
E2e tests are pretty expensive. This one is covered well enough by the webhook unit tests, so I suggest not adding it.

Reply:
Removed!